docs/se/guides/05_security_oidc.adoc

```diff
@@ -348,6 +357,81 @@ now protected, its access is limited, and the tests are not built to take oidc s
+
+Keycloak supports many authentication and authorization flows and only two of them will be shown. This chapter describes
+the different way to get an access token, refresh token and/or identity token. The identity token contains information
+about the user. The access token contains access information that the application can use to determine what resources
+the user is allowed to access. Once expired, the refresh token allows the application to obtain a new access token.
+As these token contains sensitive information, they are valid for a very short period. It is possible to make them
+last longer in order to let you manipulate them with Postman. To do so:
+
+* Go to the admin console
+* Click on the `Realm Setting` in the left menu
+* Navigate to the `Tokens` tab
+
+You can increase the access token lifespan.
+
+==== Authorization Code Flow
+
+It is a suitable flow for a browser base application. It is composed of three main steps:
+
+1. The browser visit the application. The user is not logged in, so it redirects the browser to keycloak which require
+username and password for authentication.
+2. Keycloak authenticates the user and return a temporary authorization code as a query parameter in the url.
+3. The authorization code is used to get access and refresh token from keycloak token endpoint.
+
+For the first step, open your favorite browser and copy past into the url:
+`http://localhost:8080/auth/realms/myRealm/protocol/openid-connect/auth?client_id=myClientID&response_type=code`.
+The first part of the url `http:/../auth` is the keycloak endpoint to request an authorization code. Two query
+parameters are provided, the client id and the response type.
+Press enter and Keycloak responds by a different url containing a query parameter `code`. You successfully received
+the authorization code.
+
+In order to achieve the third step, we can use Postman to exchange the authorization code for tokens. In Postman,
+select the Http POST method. Keycloak endpoint to get token is the following:
+`http://localhost:8080/auth/realms/myRealm/protocol/openid-connect/token`.
+In the body of the request, select `x-www-form-urlencoded` type. Add the following data:
+
+[source,json]
+.Enter the key:value
+----
+[{"key":"grant_type","value":"authorization_code"},
+{"key":"client_id","value":"myClientID"},
+{"key":"client_secret","value":"client secret"},
+{"key":"code","value":"authorization code"}]
+----
+
+Do not forget to replace the `client secret` by its value (generated during Create a Client), and `authorization code`
+by the code value in the query parameter. Send the request by pressing `Send`. Keycloak returns an access token and
+a refresh token.
+
+==== Resource Owner Password Credentials Grant (Direct Access Grants)
+
+The Direct Access Grants flow is used by REST clients which wants to request tokens on behalf of a user.
+Use Postman to make this request on behalf of `myuser`. Select the GET method and enter this url:
```

Comment:
To use Postman to make this request on behalf of `myuser`, select the GET method and enter this URL: